homeassistant/components/hue.py

```diff
@@ -168,6 +168,11 @@ def __init__(self, host, hass, filename, allow_unreachable=False,
 
         hass.data[DOMAIN][socket.gethostbyname(host)] = self
 
+    @property
+    def bridge_id(self):
+        """Return a unique identifier for the bridge."""
+        return socket.gethostbyname(self.host)
```

Comment:
This does I/O which is not allowed in asyncio coroutines. Below you access this property inside a coroutine. We don't expect the ip address of the bridge to change during runtime, right? So you should be able to lookup the address in this thread and save the bridge ip address in an instance attribute and return that here.

Reply:
Of course! not sure why I thought this would be a good idea. How about I just resolve in the constructor? is it OK to do I/O there?

Reply:
Actually, we already do :) I'll do that.